atomic_reactor/plugins/pre_add_image_content_manifest.py

```diff
@@ -134,13 +142,15 @@ def icm(self):
 
         :return: dict, the ICM as a Python dict
         """
-        if self.icm_url is None and self._icm is None:
+        if not self.remote_sources or (
+                self.remote_sources[0]["icm_url"] is None and self._icm is None
```

Comment:
I don't think you need to check for `remote_sources[0]["icm_url"] is None`. I think checking that `remote_sources` is not None should be good enough. You could even prevent icm_url from being None in the schema (atomic_reactor/schemas/user_params.json)

The same pattern appears in other plugins too, checking that `remote_sources` is not None should usually be good enough.

Reply:
So if remote_sources was specified, we would expect that all values will be present and not None for each remote source?

From https://github.com/containerbuildsystem/atomic-reactor/blob/master/atomic_reactor/plugins/pre_download_remote_source.py#L107

It looks like configs parameter might be empty?

Reply:
Yeah, remote_sources will be a list of objects with
```python
build_args: dict[str, str]
configs: str  # this is actually a url
request_id: int
url: str
name: Optional[str]
```
i.e. `name` is the only one which can ever be `None`